tracer/src/Datadog.Trace/TraceContext.cs

```diff
@@ -62,15 +59,15 @@ public TraceContext(IDatadogTracer tracer, TraceTagCollection? tags = null)
 
             Tracer = tracer;
             Tags = tags ?? new TraceTagCollection();
-            _clock = TraceClock.Instance;
```

Comment:
Really? I assumed that would get inlined away. I'm happy to revert this, it was just a R# suggestion, no strong feelings either way.

Reply:
Yeah, I know because I've spoken to him about it specifically 😄 In newer .NETs it's probably fine, but older pretty sure he's benchmarked and it's slower 🙂

Reply:
But... that doesn't make sense. 😕

Reply:
It's extra indirection that isn't inlined 🤷‍♂️

Reply:
I think the issue got resolved when we enabled the inlining tbh. But yeah I used to have a benchmark for that